docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,180 @@
+whenever it is updated.
+
+:::note
+If the file `/etc/tedge/plugins/tedge-log-plugin.toml` is ill-formed or cannot be read,
+then a JSON message with an empty array for the `types` field is sent, indicating no log files are tracked.
+:::
+
+## Handling log upload commands
+
+The plugin establishes an MQTT connection to the broker using the `mqtt.bind.address` and `mqtt.bind.port` values from the `tedge.toml` configuration.
+Then, it subscribes to the `<root>/<identifier>/cmd/log_upload/+` topic, listening for log file upload commands.
+The `<root>/<identifier>` is defined in `tedge.toml` file as the `device.topic` key.
+
+For instance, if `device.topic` is set to `te/device/main//`, the plugin subscribes to the following topic.
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main///cmd/log_upload/+'
+```
+
+A new log file upload command with the ID "1234" is published by another component as below.
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/log_upload/1234' '{
+  "status": "init",
+  "uploadUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/log_upload/mosquitto-1234",
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-23T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000
+}'
+```
+
+The plugin then checks the `tedge-log-plugin.toml` file for the log type in the incoming message (`mosquitto`),
+retrieves the log files using the `target` glob pattern provided in the plugin config file,
+including only the ones modified within the date range(`2013-06-22T17:03:14.000+02:00` to `2013-06-23T18:03:14.000+02:00`),
+with the content filtered by the search text(`ERROR`) and the maximum line count(`1000`).
+
+This filtered content is then uploaded to the given URL as `uploadUrl` via an HTTP PUT request.
+
+During the process, the plugin updates the command status via MQTT
+by publishing a retained message to the topic `<root>/<identifier>/cmd/log_upload/<id>`,
+which is the same topic where the command is received.
+
+On the reception of a new log file upload command, the plugin updates the status to `executing `.
+On the reception of the HTTP success status code after uploading a file to the file transfer repository, the plugin updates the status to `successful`.
+If any unexpected error occurs, the plugin updates the status to `failed` with a `reason`.
+
+Thus, the operation status update message for the above example looks like below.
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/log_upload/1234' '{
+  "status": "failed",
+  "reason": "The target log file for 'mosquitto' does not exist.",
+  "uploadUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/log_upload/mosquitto-1234",
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-22T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000
```

Comment:
Although I understand the need to keep these fields while an operation is transitioning between intermediate states like `executing` to `uploading` to `uploaded`, so that each state handler has the complete context, I'd at least expect these fields to be optional for the terminal states like `successful` and `failed`. Else, it may not be very user-friendly.

Reply:
It's up to a cloud mapper how it converts the `successful`/`failed` message and the converted outcome can be more than one messages (e.g. c8y-mapper needs to update the software list after software update operation). In my opinion, having all information in the payload and let the mapper ignore uninterested fields is more user-friendly.

Reply:
Ah I think I get your point. In short, I also think no other fields should be included for `successful`/`failed`(the final states). 
However, the current proposal here is not fully synced with the state machine proposal. In this proposal's `successful` is actually `uploaded`, but I want to keep states the simplest until we finished to define state machine. So, your point will make sense.

Reply:
> Although I understand the need to keep these fields while an operation is transitioning between intermediate states like `executing` to `uploading` to `uploaded`, so that each state handler has the complete context, I'd at least expect these fields to be optional for the terminal states like `successful` and `failed`. Else, it may not be very user-friendly.

Why not user friendly? Because too verbose?

I see these fields as useful for many small reasons: generating events and alarms for instance. It can also useful for audit purposes: an audit plugin could be implemented to record all the init and final states.

Reply:
> Why not user friendly? Because too verbose?

During the dev phase, I've always liked the simplicity of the SmartREST messages that lets you finish a dangling operation by simply sending a `503` or `504` message. Even while writing simple clients, it comes in handy. So, even for tedge, I'd appreciate the ability to finish an operation with minimal inputs like `{ "status": "successful" }`.

Of course there isn't much value in production as we'd probably recommend devs to always republish the input fields for all state transitions. That's why I was suggesting to make them optional rather than cutting them out completely.